flask_selfdoc/autodoc.py

```diff
@@ -1,19 +1,39 @@
+import json
 from operator import attrgetter, itemgetter
 import os
 import re
 from collections import defaultdict
 import sys
 import inspect
+from typing import Optional, Tuple
 
 from flask import current_app, render_template, render_template_string, jsonify
-from jinja2 import evalcontextfilter, Markup
 from jinja2.exceptions import TemplateAssertionError
 
+try:
+    # Jinja2 < 3.1 (Flask <= 2.0 and python 3.6)
+    # https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.evalcontextfilter
+    from jinja2 import evalcontextfilter as pass_eval_context
+except ImportError:
+    # Jinja2 < 3.1 (Flask >= 2.0 and python <= 3.7)
+    from jinja2 import pass_eval_context
+
+try:
+    # Jinja2 < 3.1 (Flask <= 2.0 and python 3.6)
+    from jinja2 import Markup
+except ImportError:
+    # Jinja2 < 3.1 (Flask >= 2.0 and python <= 3.7)
+    from jinja2.utils import markupsafe
+    Markup = markupsafe.Markup
 
 try:
-    from flask import _app_ctx_stack as stack
```

Comment:
With python 3.11 and Flask 2.2.3 I was getting `_app_ctx_stack' is deprecated and will be removed in Flask 2.3.`
See https://github.com/pallets/flask/pull/4682